module/zfs/zfs_acl.c

```diff
@@ -2418,12 +2421,13 @@ zfs_fastaccesschk_execute(znode_t *zdp, cred_t *cr)
 		return (0);
 	}
 
-	if (FUID_INDEX(zdp->z_uid) != 0 || FUID_INDEX(zdp->z_gid) != 0) {
+	if (zfs_uid_read(ZTOI(zdp)) != 0 ||
```

Comment:
The FUID_INDEX() macro should be preserved here to avoid a functional change.  This Solaris fuid gunk is a mess and should be reviewed and perhaps removed.